plugins/resources/compound/libcompound.cpp

```diff
@@ -696,7 +759,14 @@ irods::error repl_object(
         [&]()
         {
             if (destination_l1descInx > 0) {
+                auto [replica, lm] =
+                    irods::experimental::replica::duplicate_replica(*L1desc[destination_l1descInx].dataObjInfo);
                 close_replica(_ctx, destination_l1descInx, error_code_set_to_oprStatus_on_dest_close);
+
+                // If we got a REPLICA_IS_BEING_STAGED error, just remove the cache replica.
+                if (REPLICA_IS_BEING_STAGED == error_code_set_to_oprStatus_on_dest_close) {
+                    irods::purge_cache(*_ctx.comm(), *replica.get());
```

Comment:
Just acknowledging here that this will not work for users whose permission level on the data object does allow for unlinking. Let's ignore that case for now because I'm working in parallel to this change on #6997 to address the situation.

No action required here